compiler/rustc_passes/src/check_attr.rs

```diff
@@ -521,6 +521,103 @@ impl CheckAttrVisitor<'tcx> {
         }
     }
 
+    fn check_doc_codeblock_attr_value(
+        &self,
+        meta: &NestedMetaItem,
+        doc_codeblock_attr: &str,
+        is_list: bool,
+    ) -> bool {
+        let tcx = self.tcx;
+        let err_fn = move |span: Span, msg: &str| {
+            tcx.sess.span_err(
+                span,
+                &format!(
+                    "`#[doc(codeblock_attr{})]` {}",
+                    if is_list { "(\"...\")" } else { " = \"...\"" },
+                    msg,
+                ),
+            );
+            false
+        };
+        if doc_codeblock_attr.is_empty() {
+            return err_fn(
+                meta.name_value_literal_span().unwrap_or_else(|| meta.span()),
+                "attribute cannot have empty value",
+            );
+        }
+        if let Some(c) = doc_codeblock_attr
+            .chars()
+            .find(|&c| c == ',' || c == '`' || c == '"' || c == '\'' || c.is_whitespace())
+        {
+            self.tcx.sess.span_err(
+                meta.name_value_literal_span().unwrap_or_else(|| meta.span()),
+                &format!(
+                    "{:?} character isn't allowed in `#[doc(codeblock_attr{})]`",
+                    c,
+                    if is_list { "(\"...\")" } else { " = \"...\"" },
+                ),
+            );
+            return false;
+        }
+        if doc_codeblock_attr.starts_with(' ') || doc_codeblock_attr.ends_with(' ') {
```

Comment:
This check will never run, you already give an error above if the lang string contains whitespace.